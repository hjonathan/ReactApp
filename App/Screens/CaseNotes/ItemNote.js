import React, {Component} from "react";
import {
    Text,
    View,
    Image
} from "react-native";
import Moment from "moment";
import Card from "../../Components/Card";
import Styles from "./styles";
import * as Res from "../../Assets/resources";
import {DATETIME} from "../../Libs/Const";

class ItemNote extends Component {
    constructor (props) {
        super(props);
        this.state = {
            loaded: false
        };
        this._onLoadImage = this._onLoadImage.bind(this);
    }
    /**
     * Executes when even load was finished
     * @private
     */
    _onLoadImage = () => {
        this.setState({loaded: true});
    };

    /**
     * Gets First letter of first name
     * @returns {string}
     * @private
     */
    _getCapitalLetter (firstName) {
        const index = 0;
        return firstName.charAt(index).toUpperCase();
    }

    /**
     * Renders avatar
     * @param user
     * @returns {*}
     */
    renderImageAvatar (user) {
        let textDisplay,
            imageDisplay;
        if (this.state.loaded) {
            textDisplay = "none";
            imageDisplay = "flex";
        } else {
            textDisplay = "flex";
            imageDisplay = "none";
        }
        return (
            <View style={[Styles.circleImageContainer]}>
                <Image
                    onLoad={this._onLoadImage}
                    source={{uri: `data:image/gif;base64,${user.userPhoto}`}}
                    style={[Styles.circleImage, {display: imageDisplay}]}
                />
                <Text style={{color: "white", display: textDisplay}}>{this._getCapitalLetter(user.user.firstName)}</Text>
            </View>
        );
    }

    /**
     * Renders item list
     * @returns {*}
     */
    render () {
        const {data} = this.props;
        if (data) {
            return (
                <Card>
                    <View style={Styles.firstSection}>
                        <View>
                            <Text style={{color: "black"}}>{data.notes.content}</Text>
                        </View>
                    </View>
                    <View style={Styles.dividerLine} />
                    <View style={Styles.secondSection}>
                        <View style={Styles.leftAlign}>
                            {this.renderImageAvatar(data)}
                            <View style={{flex: 1, flexDirection: "row"}}>
                                <Text style={Styles.userName}>
                                    {data.user.firstName} {data.user.lastName}
                                </Text>
                            </View>
                        </View>
                        <View style={Styles.rightAlign}>
                            <Text style={{color: Res.colors.lightColor}}>
                                {Moment(data.notes.date).format(DATETIME.LOCAL)}
                            </Text>
                        </View>
                    </View>
                </Card>
            );
        }
        return null;
    }
}

export default ItemNote;
