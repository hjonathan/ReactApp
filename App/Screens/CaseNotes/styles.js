import {StyleSheet} from "react-native";
import * as Res from "../../Assets/resources";

export default StyleSheet.create({
    mainContainer: {
        flex: 1,
        flexDirection: "row",
        paddingVertical: 8,
        paddingHorizontal: 20,
        backgroundColor: "#2B6697"
    },
    firstSection: {
        flexDirection: "row",
        padding: 8
    },
    secondSection: {
        flex: 1,
        alignItems: "center",
        flexDirection: "row",
        padding: 8
    },
    dividerLine: {
        flex: 1,
        backgroundColor: Res.colors.gray,
        marginHorizontal: 8,
        height: 1
    },
    circleImage: {
        borderRadius: 100,
        width: 36,
        height: 36,
        resizeMode: "cover"
    },
    circleImageContainer: {
        backgroundColor: Res.colors.mainColor,
        borderRadius: 50,
        width: 36,
        height: 36
    },
    userName: {
        flexWrap: "wrap",
        paddingHorizontal: 4,
        color: Res.colors.lightColor
    },
    leftAlign: {
        flex: 1,
        flexDirection: "row",
        justifyContent: "flex-start",
        alignItems: "center",
        paddingRight: 4,
        width: "50%"
    },
    rightAlign: {
        flex: 1,
        flexDirection: "row",
        justifyContent: "flex-end",
        alignItems: "center",
        paddingLeft: 4,
        width: "50%"
    }
});
