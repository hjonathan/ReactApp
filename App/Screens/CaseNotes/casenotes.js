import React, {Component} from "react";
import {
    Text,
    TextInput,
    View,
    FlatList,
    TouchableOpacity,
    Dimensions
} from "react-native";
import Modal from "react-native-modal";
import Toast from "react-native-easy-toast";
import {CheckBox} from "react-native-elements";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";
import _ from "lodash";
import I18n from "../../I18n";
import Styles from "./styles";
import * as Res from "../../Assets/resources";
import ItemNote from "./ItemNote";

const {height} = Dimensions.get("window");

class CaseNotes extends Component {
    /**
     * Navigation option
     */
    static navigationOptions = ({navigation}) => ({
        title: I18n.t("title_activity_case_notes"),
        headerTintColor: Res.colors.white,
        headerStyle: {
            backgroundColor: Res.colors.mainColor
        },
        headerRight: (
            <TouchableOpacity
                style={{padding: 8}}
                onPress={() => {
                    navigation.state.params.handlerModal(true);
                }}
            >
                <MaterialIcon name="add" style={{color: Res.colors.white, fontSize: 24}} />
            </TouchableOpacity>)
    });
    /**
     * Constructor
     * @param props
     */
    constructor (props) {
        super(props);
        this.state = {
            checked: false,
            isEmpty: false,
            isModalVisible: false,
            toastMessage: I18n.t("text_missing_activity_case_notes_add_case_dialog"),
            errorPermission: false,
            appUid: this.props.navigation.state.params.appUid,
            notes: [],
            valid: false,
            noteContent: "",
            refreshed: false,
            middleHeight: height / 2 - 50
        };
        this.changeLayout = this.changeLayout.bind(this);
    }

    /**
     * Gets list notes.
     */
    componentWillMount () {
        this._getListNotes();
    }

    /**
     * Pass toggleModal function to navigation params
     */
    componentDidMount () {
        this.props.navigation.setParams({
            handlerModal: this._toggleModal
        });
    }

    /**
     * Component will receive props
     */
    componentWillReceiveProps (nextProps) {
        try {
            if (Object.prototype.hasOwnProperty.call(nextProps.caseNotes, "status") &&
                nextProps.caseNotes.status === "ok" &&
                !this.state.refreshed) {
                this.setState({refreshed: true});
                this._getListNotes();
            } else if (Object.prototype.hasOwnProperty.call(nextProps.caseNotes, "error") &&
                nextProps.caseNotes.error &&
                nextProps.caseNotes.error.error_description &&
                this.state.errorPermission) {
                this.refs._toastCaseNotesPost.show(nextProps.caseNotes.error.error_description, 4000);
            }
            if (Object.prototype.hasOwnProperty.call(nextProps.caseNotes, "notes") &&
                nextProps.caseNotes.notes.length && !this.state.dataRequested) {
                this.setState({dataRequested: true});
                this._getUsersInfo(nextProps.caseNotes.notes);
            }
            if (Object.prototype.hasOwnProperty.call(nextProps, "users") &&
                nextProps.users.length) {
                this.mergeData(nextProps.caseNotes);
            }
        // eslint-disable-next-line no-empty
        } catch (error) {
        }
    }

    /**
     * Gets userIds
     * @param data
     * @returns {Array}
     */
    getUserIds (data) {
        let ids = [];
        data.forEach((item) => {
            ids.push(item.user.userId);
        });
        return _.uniq(ids);
    }

    /**
     * Merges userPhoto props to dataList
     * @param data
     */
    mergeData (data) {
        let i,
            j,
            dataMerged = data.notes,
            cbUser = function (user) {
                return user.userId === dataMerged[i].user.userId;
            };
        for (i = 0; i < dataMerged.length; i += 1) {
            j = _.findIndex(data.users, cbUser);
            dataMerged[i].userPhoto = data.users[j].userPhoto;
        }
        this.setState({notes: dataMerged, isEmpty: dataMerged.length === 0});
    }

    /**
     * Gets usersInfo
     * @param data
     * @private
     */
    _getUsersInfo (data) {
        this.props.userInfoRequest({
            size: "1",
            user: {
                ids: this.getUserIds(data)
            }
        });
    }

    /**
     * Request list notes.
     * @private
     */
    _getListNotes () {
        this.props.requestCaseNotes({
            app_uid: this.state.appUid
        });
    }

    /**
     * Populate notes.
     * @param notes
     * @private
     */
    _populateNotes (notes) {
        this.setState({notes, isEmpty: (notes.length === 0)});
    }

    /**
     * Event before to cancel
     * @private
     */
    _onCancel () {
        this._toggleModal(false);
        this.setState({refreshed: false});
        this.restartModal();
    }

    /**
     * Event before to send
     * @private
     */
    _beforeToSend () {
        if (this.state.valid) {
            this._toggleModal(false);
            this.setState({refreshed: false});
        } else {
            this.refs._toastCaseNotes.show(this.state.toastMessage);
        }
        this.sendNote();
        this.restartModal();
    }

    /**
     * Validate and send new note.
     */
    sendNote = () => {
        if (this.state.valid) {
            this.setState({errorPermission: true});
            this.props.postCaseNote({
                app_uid: this.state.appUid,
                noteContent: this.state.noteContent,
                sendEmail: this.state.checked ? 1 : 0
            });
            this.setState({notes: [], isEmpty: false});
        } else {
            this.refs._toastCaseNotes.show(this.state.toastMessage, 5000);
        }
    };

    /**
     * Shows and hides dialog modal.
     * @private
     */
    _toggleModal = (state) => {
        this.setState({isModalVisible: state, checked: false});
    };

    /**
     * Restarts content modal
     */
    restartModal () {
        this.setState({noteContent: "", checked: false, valid: false});
    }

    /**
     * Updates validation when text content changes.
     * @param text
     * @private
     */
    _onChangeNote = (text) => {
        this.setState({valid: (text.trim().length > 0), noteContent: text});
    };

    /**
     * Gets new dimensions and sets middleHeight state
     */
    changeLayout () {
        this.setState({middleHeight: Dimensions.get("window").height / 2 - 50});
    }

    /**
     * Renders empty view
     * @returns {*}
     */
    emptyList () {
        return (
            <View
                style={{
                    flex: 1, justifyContent: "center", alignItems: "center", paddingTop: this.state.middleHeight
                }}
                onLayout={this.changeLayout}
            >
                <Text style={{color: "white", fontWeight: "bold"}}>{I18n.t("case_notes_no_data")}</Text>
            </View>
        );
    }

    renderList () {
        return (
            <FlatList
                data={this.state.notes}
                renderItem={({item}) => <ItemNote data={item} />}
                keyExtractor={(item, index) => index}
                ListEmptyComponent={this.emptyList()}
            />
        );
    }

    /**
     * Renders screen
     * @returns {*}
     */
    render () {
        return (
            <View style={Styles.mainContainer} onLayout={this._changeLayout}>
                {this.renderList()}
                <Modal backdropOpacity={0.6} isVisible={this.state.isModalVisible}>
                    <View style={{
                        flex: 1,
                        flexDirection: "column",
                        borderRadius: 3,
                        backgroundColor: "white"
                    }}
                    >
                        <View style={{height: 48, backgroundColor: Res.colors.lightColor}} />
                        <View style={{flex: 1, padding: 16}}>
                            <TextInput
                                style={{
                                    flex: 1,
                                    backgroundColor: "white",
                                    borderWidth: 1,
                                    borderColor: "gray",
                                    textAlignVertical: "top"
                                }}
                                onChangeText={this._onChangeNote}
                                multiline
                                placeholder={I18n.t("note_hint_activity_case_notes_route_case_dialog")}
                                underlineColorAndroid="transparent"
                            />
                            <CheckBox
                                center
                                containerStyle={{backgroundColor: "white", borderWidth: 0, padding: 0}}
                                title={I18n.t("note_send_mail_activity_case_notes_route_case_dialog")}
                                checked={this.state.checked}
                                checkedIcon="check-square"
                                checkedColor={Res.colors.successColor}
                                textStyle={{fontWeight: "normal"}}
                                onPress={() => {
                                    this.setState({checked: !this.state.checked});
                                }}
                            />
                        </View>
                        <View style={{height: 48, justifyContent: "flex-end", flexDirection: "row"}}>
                            <View style={{width: "50%"}}>
                                <TouchableOpacity
                                    onPress={() => {
                                        this._onCancel();
                                    }}
                                    style={{
                                        flex: 1,
                                        backgroundColor: Res.colors.dangerColor,
                                        alignItems: "center",
                                        justifyContent: "center"
                                    }}
                                >
                                    <Text
                                        style={{color: "white"}}
                                    >{I18n.t("cancel_activity_case_notes_route_case_dialog").toUpperCase()}
                                    </Text>
                                </TouchableOpacity>
                            </View>
                            <View style={{width: "50%"}}>
                                <TouchableOpacity
                                    onPress={() => {
                                        this._beforeToSend();
                                    }}
                                    style={{
                                        flex: 1,
                                        backgroundColor: Res.colors.successColor,
                                        alignItems: "center",
                                        justifyContent: "center"
                                    }}
                                >
                                    <Text
                                        style={{color: "white"}}
                                    >{I18n.t("post_activity_case_notes_route_case_dialog").toUpperCase()}
                                    </Text>
                                </TouchableOpacity>
                            </View>
                        </View>
                    </View>
                    <Toast
                        ref="_toastCaseNotes"
                        style={{backgroundColor: Res.colors.lightColor}}
                        textStyle={{color: "white", textAlign: "center"}}
                    />
                </Modal>
                <Toast
                    ref="_toastCaseNotesPost"
                    position="bottom"
                    positionValue={150}
                    style={{backgroundColor: Res.colors.lightColor, marginHorizontal: 16}}
                    textStyle={{color: "white", textAlign: "center"}}
                />
            </View>
        );
    }
}

export default CaseNotes;
