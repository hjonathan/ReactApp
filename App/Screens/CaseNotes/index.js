import {connect} from "react-redux";
import CaseNotes from "./casenotes";
import actions from "../../Actions";

/**
 * Return state from Store to View Component
 * @param state
 */
const mapStateToProps = state => ({
    caseNotes: state.CaseNotes || null,
    users: state.CaseNotes.users
});

const mapDispatchToProps = dispatch => ({
    requestCaseNotes: params => dispatch(actions.caseNotes.request(params)),
    postCaseNote: params => dispatch(actions.postCaseNote.request(params)),
    userInfoRequest: params => dispatch(actions.notesUsers.request(params))
});

export default connect(mapStateToProps, mapDispatchToProps)(CaseNotes);
