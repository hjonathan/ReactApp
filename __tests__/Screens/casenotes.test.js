import React from 'react'
import {shallow} from 'enzyme';
import ConnectCaseNotes from '../../App/Screens/CaseNotes';
import CaseNotes from '../../App/Screens/CaseNotes/casenotes';
import ItemNote from '../../App/Screens/CaseNotes/ItemNote';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

describe('Screen >>> Case Notes', () => {
    const initialState = {
        CaseNotes: {
            app_uid: null,
            notes: [],
            error: null,
            users: []
        }
    };
    const notes = [
        {
            caseId: "1111111111",
            notes: {
                content: "Lorem ipsum dolor",
                date: "2018-02-09T16:16:36+00:00"
            },
            user: {
                email: "john.guest@gmail.com",
                firstName: "John",
                lastName: "Guest",
                name: "John",
                userId: "123456789"
            }
        }
    ], users = [
        {
            userId: "123456789",
            userPhoto: "picture1"
        }
    ];
    const data = {
        notes: {
            content: '',
            date: ''
        },
        user: {
            firstName: '',
            lastName: ''
        }
    };
    const navigation = {
        state: {
            params: {
                handlerModal: () => {}
            }
        },
        setParams: () => {}
    };
    const toast = {
        _toastCaseNotes: {
            show: (message, duration) => {}
        },
        _toastCaseNotesPost: {
            show: (message, duration) => {}
        }
    };
    const requestCaseNotes = () => {};
    const postCaseNote = () => {};
    const userInfoRequest = jest.fn();
    const nextProps = {
        caseNotes: {
            status: 'ok',
            notes: notes
        },
        users
    };
    const nextPropsError = {
        caseNotes: {
            error: {
                error_description: "error permission"
            },
            notes: []
        }
    };
    const mockStore = configureStore();
    let store,
        container,
        view,
        itemNoteEmpty,
        itemNote;

    beforeEach(() => {
        store = mockStore(initialState);
        container = shallow(<ConnectCaseNotes store={store}/>);
        view = shallow(<CaseNotes
            navigation={navigation}
            requestCaseNotes={requestCaseNotes}
            postCaseNote={postCaseNote}
            userInfoRequest={userInfoRequest}
        />);
        itemNoteEmpty = shallow(<ItemNote/>);
        itemNote = shallow(<ItemNote data={data}/>);
    });

    it('Render Case Notes index', () => {
        expect(toJson(container)).toMatchSnapshot();
    });

    it('Check static navigationOptions', () => {
        expect(CaseNotes.navigationOptions("").title)
            .toEqual("title_activity_case_notes");
        expect(CaseNotes.navigationOptions("").headerTintColor)
            .toEqual("#FFF");
        expect(CaseNotes.navigationOptions("").headerStyle)
            .toEqual({backgroundColor: "#1F6CBC"});
        expect(CaseNotes.navigationOptions("").headerRight).toMatchSnapshot();
    });

    it('mapStateToProps', () => {
        expect(container.props('children').requestCaseNotes()).toMatchSnapshot();
        expect(container.props('children').postCaseNote()).toMatchSnapshot();
        expect(container.props('children').userInfoRequest()).toMatchSnapshot();
    });

    it('Render Case Notes view', () => {
        expect(toJson(view)).toMatchSnapshot();
    });

    it('Render Item Notes', () => {
        expect(toJson(itemNote)).toMatchSnapshot();
        expect(toJson(itemNoteEmpty)).toMatchSnapshot();
    });

    it('Render list', () => {
        view.instance().setState({notes: notes});
        let list = view.instance().renderList();
        expect(view.instance().state.notes.length).toEqual(1);
        expect(list).toMatchSnapshot();
        expect(list.props.renderItem({})).toMatchSnapshot();
        expect(list.props.keyExtractor({}, 1)).toMatchSnapshot();
    });

    it('Render list empty', () => {
        view.instance().setState({isEmpty: true});
        let list = view.instance().renderList();
        expect(view.instance().state.isEmpty).toEqual(true);
        expect(list).toMatchSnapshot();
    });

    it('componentWillReceiveProps', () => {
        view.instance().refs = toast;
        view.instance().setState({errorPermission: true});
        expect(view.instance().componentWillReceiveProps(nextProps)).toMatchSnapshot();
        expect(view.instance().componentWillReceiveProps(nextPropsError)).toMatchSnapshot();
    });

    it('_populateNotes', () => {
        let notes = [
            {
                caseId: "1111111111",
                notes: {
                    content: "Lorem ipsum dolor",
                    date: "2018-02-09T16:16:36+00:00"
                },
                user: {
                    email: "john.guest@gmail.com",
                    firstName: "John",
                    lastName: "Guest",
                    name: "John",
                    userId: "123456789"
                }
            },
            {
                caseId: "2222222222",
                notes: {
                    content: "Lorem ipsum dolor",
                    date: "2018-02-09T16:16:36+00:00"
                },
                user: {
                    email: "john.guest@gmail.com",
                    firstName: "John",
                    lastName: "Guest",
                    name: "John",
                    userId: "123456789"
                }
            }
        ];
        expect(view.instance()._populateNotes(notes)).toMatchSnapshot();
        expect(view.instance().state.notes.length).toEqual(notes.length);
        expect(view.instance().state.isEmpty).toEqual(false);
    });

    it('changeLayout', () => {
        expect(view.instance().changeLayout()).toMatchSnapshot();
    });

    it('_onChangeNote', () => {
        const message = "message test";
        expect(view.instance()._onChangeNote(message)).toMatchSnapshot();
        expect(view.instance().state.valid).toEqual(true);
        expect(view.instance().state.noteContent).toEqual(message);
    });

    it('onPress checkbox', () => {
        let modal = view.props('children').children[1].props.children[0],
            sectionBody = modal.props.children[1],
            checkbox = sectionBody.props.children[1];
        expect(checkbox.props.onPress()).toMatchSnapshot();
    });

    it('onPress cancel', () => {
        let modal = view.props('children').children[1].props.children[0],
            sectionFooter = modal.props.children[2],
            btnCancel = sectionFooter.props.children[0].props.children;
        expect(btnCancel.props.onPress()).toMatchSnapshot();
    });

    it('onPress send', () => {
        let modal = view.props('children').children[1].props.children[0],
            sectionFooter = modal.props.children[2],
            btnSend = sectionFooter.props.children[1].props.children;
        view.instance().refs = toast;
        expect(btnSend.props.onPress()).toMatchSnapshot();
        view.instance().setState({valid: true});
        expect(btnSend.props.onPress()).toMatchSnapshot();
    });

    it("Should call mergeData method", () => {
        let notesWithPicture = Object.assign(notes, users.userPhoto);
        view.instance().mergeData({notes, users});
        expect(view.instance().state.notes[0])
            .toEqual(notesWithPicture[0]);
    });
});
