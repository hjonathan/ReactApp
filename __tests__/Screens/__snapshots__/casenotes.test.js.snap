// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen >>> Case Notes _onChangeNote 1`] = `undefined`;

exports[`Screen >>> Case Notes _populateNotes 1`] = `undefined`;

exports[`Screen >>> Case Notes Check static navigationOptions 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "padding": 8,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="add"
    size={12}
    style={
      Object {
        "color": "#FFF",
        "fontSize": 24,
      }
    }
  />
</TouchableOpacity>
`;

exports[`Screen >>> Case Notes Render Case Notes index 1`] = `
<CaseNotes
  caseNotes={
    Object {
      "app_uid": null,
      "error": null,
      "notes": Array [],
      "users": Array [],
    }
  }
  postCaseNote={[Function]}
  requestCaseNotes={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  userInfoRequest={[Function]}
  users={Array []}
/>
`;

exports[`Screen >>> Case Notes Render Case Notes view 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#2B6697",
      "flex": 1,
      "flexDirection": "row",
      "paddingHorizontal": 20,
      "paddingVertical": 8,
    }
  }
>
  <FlatList
    ListEmptyComponent={
      <Component
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingTop": 617,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          case_notes_no_data
        </Text>
      </Component>
    }
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.6}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    hideModalContentWhileAnimating={false}
    isVisible={false}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    useNativeDriver={false}
  >
    <Component
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 3,
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Component
        style={
          Object {
            "backgroundColor": "#3397E1",
            "height": 48,
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
            "padding": 16,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="note_hint_activity_case_notes_route_case_dialog"
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "gray",
              "borderWidth": 1,
              "flex": 1,
              "textAlignVertical": "top",
            }
          }
          underlineColorAndroid="transparent"
        />
        <CheckBox
          center={true}
          checked={false}
          checkedColor="#1FBC99"
          checkedIcon="check-square"
          containerStyle={
            Object {
              "backgroundColor": "white",
              "borderWidth": 0,
              "padding": 0,
            }
          }
          iconRight={false}
          onPress={[Function]}
          right={false}
          size={24}
          textStyle={
            Object {
              "fontWeight": "normal",
            }
          }
          title="note_send_mail_activity_case_notes_route_case_dialog"
          uncheckedColor="#bfbfbf"
          uncheckedIcon="square-o"
        />
      </Component>
      <Component
        style={
          Object {
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "flex-end",
          }
        }
      >
        <Component
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E14333",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              CANCEL_ACTIVITY_CASE_NOTES_ROUTE_CASE_DIALOG
            </Text>
          </TouchableOpacity>
        </Component>
        <Component
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1FBC99",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              POST_ACTIVITY_CASE_NOTES_ROUTE_CASE_DIALOG
            </Text>
          </TouchableOpacity>
        </Component>
      </Component>
    </Component>
    <Toast
      fadeInDuration={500}
      fadeOutDuration={500}
      opacity={1}
      position="bottom"
      positionValue={120}
      style={
        Object {
          "backgroundColor": "#3397E1",
        }
      }
      textStyle={
        Object {
          "color": "white",
          "textAlign": "center",
        }
      }
    />
  </ReactNativeModal>
  <Toast
    fadeInDuration={500}
    fadeOutDuration={500}
    opacity={1}
    position="bottom"
    positionValue={150}
    style={
      Object {
        "backgroundColor": "#3397E1",
        "marginHorizontal": 16,
      }
    }
    textStyle={
      Object {
        "color": "white",
        "textAlign": "center",
      }
    }
  />
</Component>
`;

exports[`Screen >>> Case Notes Render Item Notes 1`] = `
<CardView>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "padding": 8,
      }
    }
  >
    <Component>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
          }
        }
      />
    </Component>
  </Component>
  <Component
    style={
      Object {
        "backgroundColor": "#BABABA",
        "flex": 1,
        "height": 1,
        "marginHorizontal": 8,
      }
    }
  />
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "padding": 8,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingRight": 4,
          "width": "50%",
        }
      }
    >
      <Component
        style={
          Array [
            Object {
              "backgroundColor": "#1F6CBC",
              "borderRadius": 50,
              "height": 36,
              "width": 36,
            },
          ]
        }
      >
        <Image
          onLoad={[Function]}
          source={
            Object {
              "uri": "data:image/gif;base64,undefined",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 100,
                "height": 36,
                "resizeMode": "cover",
                "width": 36,
              },
              Object {
                "display": "none",
              },
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "display": "flex",
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#3397E1",
              "flexWrap": "wrap",
              "paddingHorizontal": 4,
            }
          }
        >
           
        </Text>
      </Component>
    </Component>
    <Component
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingLeft": 4,
          "width": "50%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#3397E1",
          }
        }
      >
        Invalid date
      </Text>
    </Component>
  </Component>
</CardView>
`;

exports[`Screen >>> Case Notes Render Item Notes 2`] = `""`;

exports[`Screen >>> Case Notes Render list 1`] = `
<FlatList
  ListEmptyComponent={
    <Component
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingTop": 617,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        case_notes_no_data
      </Text>
    </Component>
  }
  data={
    Array [
      Object {
        "caseId": "1111111111",
        "notes": Object {
          "content": "Lorem ipsum dolor",
          "date": "2018-02-09T16:16:36+00:00",
        },
        "user": Object {
          "email": "john.guest@gmail.com",
          "firstName": "John",
          "lastName": "Guest",
          "name": "John",
          "userId": "123456789",
        },
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`Screen >>> Case Notes Render list 2`] = `<ItemNote />`;

exports[`Screen >>> Case Notes Render list 3`] = `1`;

exports[`Screen >>> Case Notes Render list empty 1`] = `
<FlatList
  ListEmptyComponent={
    <Component
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingTop": 617,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        case_notes_no_data
      </Text>
    </Component>
  }
  data={Array []}
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`Screen >>> Case Notes changeLayout 1`] = `undefined`;

exports[`Screen >>> Case Notes componentWillReceiveProps 1`] = `undefined`;

exports[`Screen >>> Case Notes componentWillReceiveProps 2`] = `undefined`;

exports[`Screen >>> Case Notes mapStateToProps 1`] = `
Object {
  "type": "caseNotes/request",
}
`;

exports[`Screen >>> Case Notes mapStateToProps 2`] = `
Object {
  "type": "postCaseNote/request",
}
`;

exports[`Screen >>> Case Notes mapStateToProps 3`] = `
Object {
  "type": "notesUsers/request",
}
`;

exports[`Screen >>> Case Notes onPress cancel 1`] = `undefined`;

exports[`Screen >>> Case Notes onPress checkbox 1`] = `undefined`;

exports[`Screen >>> Case Notes onPress send 1`] = `undefined`;

exports[`Screen >>> Case Notes onPress send 2`] = `undefined`;
